src/aliceVision/calibration/checkerDetector.cpp

```diff
@@ -1169,6 +1193,58 @@ bool CheckerDetector::mergeCheckerboards()
     using CheckerBoardWithScore = std::pair<CheckerBoard, double>;
     std::vector<CheckerBoardWithScore> checkers;
 
+    // Remove empty borders
+    for (int id = 0; id < _boards.size(); id++)
+    {
+        auto b = _boards[id];
+
+        int miny = b.rows() - 1;
+        int maxy = 0;
+        int minx = b.cols() - 1;
+        int maxx = 0;
+
+        for (int i = 0; i < b.rows(); i++)
+        {
+            bool empty = true;
+            for (int j = 0; j < b.cols(); j++)
+            {
+                if (b(i, j) != UndefinedIndexT)
+                {
+                    empty = false;
+                }
+            }
+
+            if (!empty)
+            {
+                miny = std::min(miny, i);
+                maxy = std::max(maxy, i);
+            }
+        }
+
+        for (int j = 0; j < b.cols(); j++)
+        {
+            bool empty = true;
+            for (int i = 0; i < b.rows(); i++)
+            {
+                if (b(i, j) != UndefinedIndexT)
+                {
+                    empty = false;
+                }
+            }
+
+            if (!empty)
+            {
+                minx = std::min(minx, j);
+                maxx = std::max(maxx, j);
+            }
+        }
```

Comment:
I agree with codefactor :-)
I'd split the function with multiple helper functions to make it more modular and readable